sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -161,11 +161,18 @@ class DataFrame private[sql](
   }
 
   /**
+    * Compose the string representing rows for output
+    */
+  def showString(): String = {
+    showString(20);
+  }
+
+  /**
    * Compose the string representing rows for output
    * @param _numRows Number of rows to show
    * @param truncate Whether truncate long strings and align cells right
    */
-  private[sql] def showString(_numRows: Int, truncate: Boolean = true): String = {
+  def showString(_numRows: Int, truncate: Boolean = true): String = {
     val numRows = _numRows.max(0)
```

Comment:
another minor thing: now that this is a public API we should expose a better parameter name
```
def showString(numRows: Int, ...): String = {
  val _numRows = numRows.max(0)
}
```